connectors/src/connectors/google_drive/temporal/file.ts

```diff
@@ -319,11 +323,59 @@ export async function syncOneFile(
                 {
                   error: err,
                 },
-                `Error while converting pptx document to text`
+                "Error while converting pptx document to text"
               );
               return false;
             }
           }
+        } else if (file.mimeType === "text/csv") {
+          if (res.data instanceof ArrayBuffer) {
+            // If data is > 4 times the limit, we skip the file since even if
+            // converted to utf-8 it will overcome the limit enforced below. This
+            // avoids operations on very long text files, that can cause
+            // Buffer.toString to crash if the file is > 500MB
+            if (res.data.byteLength > 4 * maxDocumentLen) {
+              localLogger.info({}, "File too big to be chunked. Skipping");
+              return false;
+            }
+            const tableCsv = Buffer.from(res.data).toString("utf-8").trim();
+            const tableId = file.id;
+            const tableName = slugify(file.name.substring(0, 32));
+            const tableDescription = `Structured data from Google Drive (${file.name})`;
+
+            try {
+              await upsertTableFromCsv({
+                dataSourceConfig,
+                tableId,
+                tableName,
+                tableDescription,
+                tableCsv,
```

Comment:
Follow up items, we will need to at least `slugify` the headers. You can use `getSanitizedHeaders`.

Reply:
@flvndvd shouldn't we perform this sanitisation step in the endpoint directly?